src/groupeddataframe/splitapplycombine.jl

```diff
@@ -1027,26 +1174,35 @@ function _combine_with_first(first::Union{NamedTuple, DataFrameRow, AbstractData
     elseif first isa DataFrameRow
         n = length(gd)
         eltys = [eltype(parent(first)[!, i]) for i in parentcols(index(first))]
-    else # NamedTuple giving a single row
+    elseif firstmulticol == Val(false) && first.x1 isa Union{AbstractArray{<:Any, 0}, Ref}
+        extrude_case = true
+        first = wrap_row(first.x1, firstmulticol)
+        n = length(gd)
+        eltys = (typeof(first.x1),)
+    else # other NamedTuple giving a single row
         n = length(gd)
         eltys = map(typeof, first)
         if any(x -> x <: AbstractVector, eltys)
             throw(ArgumentError("mixing single values and vectors in a named tuple is not allowed"))
         end
     end
-    idx = Vector{Int}(undef, n)
+    if isnothing(idx_agg)
+        idx = Vector{Int}(undef, n)
+    else
+        idx = idx_agg
+    end
```

Comment:
This function is going to be recompiled every time anyway, because of changes in `first` type and `f` type (i.e. different type of return value and different type of function will cause recompilation). We decided to leave `@nospecialize` for later, so I would leave it as is for now here.